src/citiesMap.cpp

```diff
@@ -0,0 +1,143 @@
+#include "citiesMap.h"
+#include <iostream>
+#include <string>
+#include <map>
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+#include <vector>
+ 
+
+void citiesMap::insert(std::string cityName, double x, double y)
+{	 
+	coordinate newCity = { x, y };
+
+	this->cityMap.insert({ cityName,newCity });
+}
+
+ 
+
+
+
+ 
+
+
+double calculateEuclideanDistance(double x0, double y0, double x1, double y1) {
+    return std::sqrt(std::pow(x0 - x1, 2) + std::pow(y0 - y1, 2));
+}
+
+double calculateChebyshevDistance(double x0, double y0, double x1, double y1) {
+    return std::max(std::abs(x0 - x1), std::abs(y0 - y1));
+}
+
+double calculateManhattanDistance(double x0, double y0, double x1, double y1) {
+    return std::abs(x0 - x1) + std::abs(y0 - y1);
+}
+
+void citiesMap::initilizedistanceFunctions()
+{
+     this->distanceFunctions = {
+        {0, calculateEuclideanDistance},
+        {1, calculateChebyshevDistance},
+        {2, calculateManhattanDistance}
+    };
+}
+
+
+
+void citiesMap::readCitiesFromFile(std::string filename)
+{
+   
+    std::ifstream file(filename);
+    std::string line;
+    std::string name;
+    double x, y;
+
+    while (std::getline(file, line)) {
+        name = line;
+
+        if (!std::getline(file, line)) {
+            throw std::runtime_error("Invalid data format in file: " + filename);
+        }
+
+        std::istringstream iss(line);
+        std::string xStr, yStr;
+        if (!(std::getline(iss, xStr, '-') && std::getline(iss, yStr))) {
+            throw std::runtime_error("Invalid data format in file: " + filename);
+        }
+
+        std::istringstream xIss(xStr);
+        std::istringstream yIss(yStr);
+        if (!(xIss >> x) || !(yIss >> y)) {
+            throw std::runtime_error("Invalid data format in file: " + filename);
+        }
+
+        this->insert(name, x, y);
+    }
+
+    
+}
+
+bool inSquare(double x0, double y0, double x1, double y1,double radius) {
+    return   std::max(std::abs(x0 - x1), std::abs(y0 - y1)) <= radius;
```

Comment:
wrong implementation of finding the square, you are calculating Chebyshev Distance and checking if the distance is smaller than the radius